File: sdks/csharp/Renamed/Exceptions/ValidationException.cs
namespace Renamed.Sdk.Exceptions;

/// <summary>
/// Exception thrown when request validation fails.
/// </summary>
public class ValidationException : RenamedExceptionBase
{
    private const string ErrorCode = "VALIDATION_ERROR";
    private const int HttpStatusCode = 400;

    /// <summary>
    /// Creates a new instance of ValidationException with a custom message.
    /// </summary>
    /// <param name="message">The error message.</param>
    /// <param name="details">Additional validation error details.</param>
    public ValidationException(string message, object? details = null)
        : base(message, ErrorCode, HttpStatusCode, details)
    {
    }

    /// <summary>
    /// Creates a new instance of ValidationException with a custom message and inner exception.
    /// </summary>
    /// <param name="message">The error message.</param>
    /// <param name="innerException">The inner exception.</param>
    /// <param name="details">Additional validation error details.</param>
    public ValidationException(string message, Exception innerException, object? details = null)
        : base(message, ErrorCode, innerException, HttpStatusCode, details)
    {
    }
}

File: sdks/php/src/Exceptions/ValidationException.php
<?php

declare(strict_types=1);

namespace Renamed\Exceptions;

/**
 * Validation error - invalid request parameters.
 */
class ValidationException extends RenamedExceptionBase
{
    public function __construct(string $message, mixed $details = null)
    {
        parent::__construct($message, 'VALIDATION_ERROR', 400, $details);
    }
}
